conformance/lang/expressions/literals/nil_literal.balt

```diff
@@ -0,0 +1,261 @@
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    () x = null;
+    io:println(x.toBalString()); // @output ()
+
+    io:println(null.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with toString().
+Labels: nil-literal, null
+
+public function main() {
+    () x = ();
+    io:println(x); // @output
+
+    io:println(().toString()); // @output
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with toString().
+Labels: nil-literal, null
+
+public function main() {
+    () x = null;
+    io:println(x); // @output
+
+    io:println(null.toString()); // @output
+}
+
+Test-Case: output
+Description: Test equality of the nil literals.
+Labels: equality, nil-literal, null, value:toBalString
+
+public function main() {
+    io:println(null == ()); // @output true
+
+    io:println(null != ()); // @output false
+}
+
+Test-Case: output
+Description: Test exact-equality of the nil literalS.
+Labels: exact-equality, nil-literal, null, value:toBalString
+
+public function main() {
+    io:println(null === ()); // @output true
+
+    io:println(null !== ()); // @output false
+}
+
+Test-Case: output
+Description: Test nil literal with is-expr.
+Labels: is-expr, nil-literal, null
+
+public function main() {
+    io:println(() is ()); // @output true
+
+    io:println(null is ()); // @output true
+}
+
+Test-Case: output
+Description: Test different nil literal representations with nil type descriptor.
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    () x = ( );
+    io:println(x.toBalString()); // @output ()
+
+    x = (    );
+    io:println(x.toBalString()); // @output ()
+
+    x = (
+    );
+    io:println(x.toBalString()); // @output ()
+
+    x = (());
+    io:println(x.toBalString()); // @output ()
+
+    x = (null);
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal '()' with function call.
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    testNil(());
+}
+
+function testNil(() x) {
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with function call.
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    testNil(null);
+}
+
+function testNil(() x) {
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal '()' with return statement.
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    io:println(testNil().toBalString()); // @output ()
+}
+
+function testNil() returns () {
+    return ();
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with return statement.
+Labels: nil-literal, null, value:toBalString
+
+public function main() {
+    io:println(testNil().toBalString()); // @output ()
+}
+
+function testNil() returns () {
+    return null;
+}
+
+Test-Case: output
```

Comment:
Shall we also add simple JSON tests with both nil and null, because the section on the nil literal specifically mentions JSON and using null with JSON?

Reply:
will add.